pkg/controller/bypass/bypass_controller.go

```diff
@@ -0,0 +1,277 @@
+package bypass
+
+import (
+	"bytes"
+	"context"
+	"embed"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path"
+	"strings"
+	"time"
+
+	"k8s.io/apimachinery/pkg/types"
+
+	netns "github.com/containernetworking/plugins/pkg/ns"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+
+	nd "istio.io/istio/cni/pkg/nodeagent"
+	"istio.io/istio/pkg/util/sets"
+	"kmesh.net/kmesh/pkg/logger"
+	"kmesh.net/kmesh/pkg/utils"
+)
+
+var (
+	log = logger.NewLoggerField("bypass")
+	FS  embed.FS
+)
+
+func getCurrentNodeName(client kubernetes.Interface, podName, namespace string) (string, error) {
+	pod, err := client.CoreV1().Pods(namespace).Get(context.Background(), podName, metav1.GetOptions{})
+	if err != nil {
+		return "", fmt.Errorf("error getting pod %s in namespace %s: %v", podName, namespace, err)
+	}
+
+	nodeName := pod.Spec.NodeName
+
+	return nodeName, nil
+}
+
+func StartPodEventWatcher(client kubernetes.Interface, podName, podNamespace string) error {
+	stopChan := make(chan struct{})
+
+	nodeName, err := getCurrentNodeName(client, podName, podNamespace)
+	if err != nil {
+		log.Error(err)
+		return err
+	}
+
+	pod, err := client.CoreV1().Pods(podNamespace).Get(context.TODO(), podName, metav1.GetOptions{})
```

Comment:
You need to get the pod once more, so why do you need the getCurrentNodeName() function?
You can Get Pod first and then get the NodeName. this will reduce the number of times you have to request the kubeclient.